src/utils/getPreparedPeople.ts

```diff
@@ -0,0 +1,62 @@
+import { Person } from '../types';
+
+function filterByName(person: Person, query: string) {
+  const normalizedQuery = query.toLowerCase();
+
+  return person.name.toLowerCase().includes(normalizedQuery)
+    || person.fatherName?.toLowerCase().includes(normalizedQuery)
+    || person.motherName?.toLowerCase().includes(normalizedQuery);
+}
+
+function calculationCentry(year: number) {
+  return Math.ceil(year / 100);
+}
+
+export function getPreparedPeople(
+  people: Person[],
+  query: string,
+  sex: string | null,
+  centuries: string[],
+  sortField: string | null,
+  order: string | null,
+) {
+  let copyPeople = [...people];
+
+  if (query) {
+    copyPeople = copyPeople.filter(person => filterByName(person, query));
+  }
+
+  if (sex) {
+    copyPeople = copyPeople.filter(person => {
+      return person.sex === sex;
+    });
+  }
+
+  if (centuries.length) {
+    const preparedCentries = centuries.map(Number);
+
+    copyPeople = copyPeople
+      .filter(({ born }) => preparedCentries.includes(calculationCentry(born)));
+  }
+
+  if (sortField) {
```

Comment:
Your function looks to complicated, you can split logic, one function that `filter` an array and the second one that sort